### PR TITLE
Display all project resources for ROOT admin in default view

### PR DESCRIPTION
**Pull request by ravening**
### Description

Currently for ROOT admin, all the project resources are displayed
only under the respective project view. It is handy to display
all project resources of all subdomains under default view for
ROOT admin

Also display domain name in the instances overview

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### Screenshots (if appropriate):

Screenshot 2021-07-14 at 12 16 33

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
> Currently for ROOT admin, all the project resources are displayed
> only under the resprctive project view. It is handy to display
> all project resources of all subdomains under default view for
> ROOT admin

@ravening , i can see why you want this, but it is a functional change and I would have it guarded by a setting or a parameter, either of which defaults to the old behaviour.

**Comment by mib1185** [comment]
Maybe this will make #4828 no longer needed 🤔

**Comment by ravening** [comment]
> > Currently for ROOT admin, all the project resources are displayed
> > only under the resprctive project view. It is handy to display
> > all project resources of all subdomains under default view for
> > ROOT admin
> 
> @ravening , i can see why you want this, but it is a functional change and I would have it guarded by a setting or a parameter, either of which defaults to the old behaviour.

@DaanHoogland i can add a global setting for it

File: server/src/main/java/com/cloud/user/AccountManagerImpl.java [hunk 1]
```diff
@@ -2696,6 +2696,13 @@ public void buildACLSearchParameters(Account caller, Long id, String accountName
             if (id == null) {
                 domainIdRecursiveListProject.third(Project.ListProjectResourcesCriteria.SkipProjectResources);
             }
+
+            if ((caller.getType() == Account.ACCOUNT_TYPE_ADMIN) ||
+                    (caller.getType() == Account.ACCOUNT_TYPE_DOMAIN_ADMIN) ||
+                    (caller.getType() == Account.ACCOUNT_TYPE_RESOURCE_DOMAIN_ADMIN) ||
+                    (caller.getType() == Account.ACCOUNT_TYPE_READ_ONLY_ADMIN)) {
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/user/AccountManagerImpl.java:2703]: We could extract all this enums to a constant list and verify if it contains the caller.getType.

**Comment by mib1185** [comment]
@PaulAngus had a good point with https://github.com/apache/cloudstack/pull/4828#issuecomment-919990854

**Comment by ravening** [comment]
closing this pr as the backend functionality already exists and ui functionality is added in #4828

**Pull request closed without merging**